server/src/main/java/org/spine3/server/BoundedContext.java

```diff
@@ -227,25 +231,56 @@ private Response validateCommand(Command command) {
      * Validates the incoming command message.
      *
      * @param commandMessage the command message to validate
-     * @return {@link Response} with {@code ok} value if the command is valid, or
-     *          with {@link org.spine3.base.Error} value otherwise
+     * @return a response with {@code OK} value if the command is valid, or
+     *          with {@link org.spine3.base.Error}/{@link Failure} value otherwise
      */
-    protected Response validateMessage(Message commandMessage) {
+    protected Response validateCommandMessage(Message commandMessage) {
         final Response result = commandBus.validate(commandMessage);
         return result;
     }
 
     @Override
-    public void notify(Event event, StreamObserver<Response> responseObserver) {
+    public void notify(IntegrationEvent integrationEvent, StreamObserver<Response> responseObserver) {
         try {
-            eventBus.post(event);
-            responseObserver.onNext(Responses.ok());
+            final Message message = fromAny(integrationEvent.getMessage());
+            final Response response = validateIntegrationEventMessage(message);
+            responseObserver.onNext(response);
             responseObserver.onCompleted();
+            if (isOk(response)) {
+                final Event event = toEvent(integrationEvent);
+                eventBus.post(event);
+            }
         } catch (RuntimeException e) {
             responseObserver.onError(e);
         }
     }
 
+    /**
+     * Validates an incoming integration event message.
+     *
+     * @param eventMessage a message to validate
+     * @return a response with {@code OK} value if the command is valid, or
+     *          with {@link org.spine3.base.Error}/{@link Failure} value otherwise
+     */
+    protected Response validateIntegrationEventMessage(Message eventMessage) {
+        final Response response = eventBus.validate(eventMessage);
+        return response;
+    }
+
+    private static Event toEvent(IntegrationEvent integrationEvent) {
+        final IntegrationEventContext sourceContext = integrationEvent.getContext();
+        final EventContext context = EventContext.newBuilder()
+                .setEventId(sourceContext.getEventId())
+                .setTimestamp(sourceContext.getTimestamp())
+                .setProducerId(sourceContext.getProducerId())
+                .setBoundedContextName(sourceContext.getBoundedContextName())
```

Comment:
I think the `producerId` is the name of the bounded context, which sent us the event. And we should not have the `boundedContextName` attribute for the regular events at all. *All of them* (apart from integration events) are going to be within this bounded context.